### ISSUE TITLE
A lot of games with issues, The core is very broken now

### ISSUE THREAD

**Issue by thiagobarros111** [libretro/FreeIntv#39]
The worst part is the best games are not working well or are unplayable:
BumpNJump and Burgertime the physics are not working as expected,
Heman, Frogger, Frog Bog,  NASL Soccer, Maze a tron are impossible to even start, and many others.
Well  its a mess, but the older versions of this core were working better, maybe can be a good idea for now revert to the older releases until a better working version becomes avaliable.

**Comment by Keltron3030** [comment]
Yeah, it's a bummer.  Maybe try PlayStation to get your intellivision fix for now.  https://retroachievements.org/game/13908

**Comment by thiagobarros111** [comment]
To tell the truth I got an Intellivision emulator for the old wii and I think I will try to get older versions of this core on libretro until these problems become fixed

**Comment by thiagobarros111** [comment]
I found it. Retroarch 1.4 is the best version of this core. Not perfect but very playable for the broken games

**Comment by recompileorg** [comment]
Things are improving. Burger Time seems fine to me.  BumpNJump is still off.  He-Man appears to be working.  Frogger has improved, it's playable, but still has some serious graphical issues.  Frog bog scoring is still broken.  I can't tell if NASL Soccer is working or not.  I never could get the hang of those sorts of games.  Scrolling seems fixed across the board though.

**Comment by Darknior** [comment]
WOAW so cool to see some one trying to fix this great emulator of a system so forgotten ...
Thanks @recompileorg :D

**Comment by Maks7878** [comment]
I'll start by saying, yes a big thank you to @recompileorg for working on this project, for a beloved and forgotten system. I wanted to say that I have persistent issues with graphic glitching with Advanced Dungeon & Dragons Cloudy Mountain 1982. While moving/scrolling thru the level the screen glitches and tears quite significantly. Enough that it is nearly impossible to play. I was wondering if this was known or being worked on. Here's a copy of my log file https://pastebin.com/Szb1wRFG

**Comment by recompileorg** [comment]
Thanks.  Yes, the scrolling problem on AD&D and Dracula is well-known and baffling.  I'll spend some time with it today, but I can't make any guarantees.

**Comment by jonaniano** [comment]
@recompileorg any luck on the scrolling issues in AD&D? I've spent some time trying to pull apart the screen drawing without much luck. If you have an idea where the problem might be, I'll try to chase it down. 

Also, huge thanks for working on this. If you have a Patreon or something let us know.

**Comment by recompileorg** [comment]
@jonaniano  No luck yet, but I haven't had much time to work on it this year.  I've had very little time to work on projects like this, oddly enough, for the same reason that I expected a lot of free time this year! 

The problem should be in stic.c -- that's where all of the video output happens.   I rewrote it  some time ago to make the code easier to follow and to, hopefully, fix some subtle bugs.  It helped a bit, and now can render per scanline rather than a whole frame at a time.   

This change also let me separate the horizontal and vertical offsets from the rest of the rendering code.  Now, each drawing routine draws a scanline (to a buffer) as though there were no offsets.  The offsets are applied when the scanline is copied to the frame buffer.  

I should probably add more elaborate comments to explain what the code does and why.   Though you can find a lot of the terms defined in this document:  
http://spatula-city.org/~im14u2c/intv/jzintv-1.0-beta3/doc/programming/stic.txt

Now, I actually maintain two versions of the emulator.  The one here, and the one I use for testing, written in JavaScript.  (I actually wrote the JS version first, with the expectation that I'd later port it to C.  As a result, almost all of the code is identical between the two versions.)  As you can imagine, this makes it easier to debug.   Both versions share the same scrolling issue.  

I've gone frame by frame and scanline by scanline and I have absolutely know idea why that "jumping" happens.   I can tell you it is subtle and seems related in some way to the horizontal and vertical delay.  Those values look correct at every point, but occasionally the rendering is wrong anyway and we get a "jump".  

You'd expect to see the "jump" happen only when the vertical or horizontal delay hits one cards width or height and wraps around, but that doesn't seem to be the case.   

I don't remember for sure, but I also think I checked to see if I was drawing a frame when the display wasn't enabled.    

I wish I had more time to dig in to it, but this year refuses to slow down.   Any help you can offer in tracking down this bizarre bug is very welcome.  Just let me know if there's anything I can answer for you and I'll try my best to make some extra time.

**Comment by Keltron3030** [comment]
To avoid creating an issue for every game that has a problem I will add here that B-17 Bomber crashes Retroarch shortly after leaving the title card.  I understand that intellivoice is not currently supported but that seems to be unrelated to this issue since other intellivoice titles do not crash and I know from playing on real hardware there is no such problem playing the game without the intellivoice module.  If it's preferred to create a separate issue for perhaps a unique issue unrelated to gameplay let me know and I will happily create one.

**Comment by Keltron3030** [comment]
Wanted to point out more games with collision problems:

Space Armada - Has an collision issue where the shots occasionally don't exit the cannon properly, as if they are hitting a bunker despite a clear shot.  There are also times when you seems to get destroyed inappropriately.

Star Strike - The game is supposed to be won when you destroy each of the 5 red targets.  The game now allows you to log multiple hits on a single pass of a single target.  If you time your bombs right you can now win the game in a single pass.  Due to this issue, most of the time you win in 2-3 shots instead of the correct 5.

Astrosmash - The least problematic of the three I've mentioned but there are frequent occurrences when multiple MOBs are destroyed inappropriately.

Conclusion: All of these issues seem to be systemic and unfortunately it seems that the core still has some fundamental flaw(s) that manifests itself differently depending on the particular game mechanic at hand.  I wish I could offer more help than just bug reporting but regrettably I don't have the technical knowledge to do more.

**Comment by JubalHarshaw** [comment]
Add Tron Deadly Discs (and Deadly Dogs) to the collision issue.

**Comment by Keltron3030** [comment]
@thiagobarros111 @Maks7878 @JubalHarshaw, there have been countless refinements to this core since you've reported problems. I can confirm all the games I've mentioned now work perfectly.  Would you guys mind testing these and confirming behavior so we can clean up old issues?

Thanks again to @nanochess for all the latest work and to @recompileorg for getting this core going in the first place. 🙂👍

**Comment by JubalHarshaw** [comment]
Tested just now after updating the core, discs still do not make contact with enemies (though the doors can be hit) on both TRON Deadly Discs and Deadly Dogs. RetroArch 1.9.14 - FreeIntv 1.2 .... I have verified my ROMs are good; I don't have any problems using them in jzintv. If there is any info I can provide to help, please let me know.

**Comment by jonaniano** [comment]
I can confirm the scrolling issue on AD&D is now fixed for me!! Appreciate all the work. Thanks @recompileorg and @nanochess

**Comment by nanochess** [comment]
> Tested just now after updating the core, discs still do not make contact with enemies (though the doors can be hit) on both TRON Deadly Discs and Deadly Dogs. RetroArch 1.9.14 - FreeIntv 1.2 .... I have verified my ROMs are good; I don't have any problems using them in jzintv. If there is any info I can provide to help, please let me know.

This should be corrected in the next pull request (already sent), because there were problems in collision detection for "thin" sprites. I didn't mentioned it because I didn't knew it was used in TRON Deadly Discs.

**Comment by madvpro** [comment]
Hi folks! I'm using FreeIntv 1.1 on recalbox 8.1.1 and I can't make the virtual keyboard appear, even pressing L or R on the gamepad. I'm using the Playsytation 5 wireless gamepad. Can anybody help me? Games that allow you to select 1 or 2 players (Burgertime, Shark Shark, etc...) are impossible to start because I don't have access to the virtual keyboard.
